### ISSUE TITLE
viewing orders belonging to store_id=2 gives "undefined error" - "not found"

### ISSUE THREAD

**Issue by RichardThomas1** [opencart/opencart#4420]
This bug relates to both master downloaded 21/05/16 (giving "error undefined") and replicated today on latest master (although now says "not found").

An order belonging to the second additional store (store_id=2) gives a "not found" error when attempting to view the order(route=sale/order/info). It works fine for orders from the default store, the first store created (store_id=1) and subsequent stores created, just not store_id=2.

As it sounds like a configuration error I have started with a fresh install from Github today and exactly the same error but only on store_id=1. I have installed a second fresh installation to replicate exactly (this time orders for all additional stores give the error):

Installed OC - created additional store "1", set stock checkout to yes, linked Canon EOS product to store "1", created user, created order (although when you select the user you have to go back to change the "store" back to "1" which is odd when the user is not assigned to a store and you have selected the store already). View the order (checking it is not a default store order) and you get "not found" error. The page renders correctly, however you cannot add a new order status in the "history" (even with all permissions set).

Not sure what actually causes the error as some stores are OKay on the first two installation and the latest installation all additional stores give an error when viewing their orders (I have tried two fresh installations and it replicates the error each time).

P.S. The "settings" menu goes direct to the default store and not the expected "stores" list (you can only get to the stores list using the breadcrumbs).

To recap:
1. Viewing  orders belonging to additional stores gives "not found" error for some stores.
2. Order history cannot be updated for stores with the above error (used to be all additional stores).
3. When adding an order you select the store then customer, which changes the store back to default.
4. "settings" goes to the default store only.

**Comment by RichardThomas1** [comment]
The "not found" error is due to index.php?route=api/login uses the storefront path saved in the order being viewed. This therefore fails if the storefront path in the order is invalid or because it has changed since the order was created (i.e. moving from test to live storefront domains) making the path invalid in existing orders.

Other pages requiring api login do not have the error as they use the admin path to the installation.

**Comment by danielkerr** [comment]
Im going to set the system to use store id when using the API, but i think there are no solutions if you delete a store. the order editor can only work correctly if you have made loads of settings changes after an order is made.

**Comment by danielkerr** [comment]
ok fixed everythign you listed.

**Comment by kumarsaurabh030** [comment]
i face same problem when i add new order select store "store1" then fill all details after click on continue button  . after it show error.
